### PR TITLE
Only disallow hyphens in lib target names

### LINKED ISSUE #1450: hyphens should be allowed in binary targets
According to https://github.com/rust-lang/cargo/pull/1443, "Binaries and examples, however, are allowed to contain hyphens in their name." However:

```
#Cargo.toml

[package]

name = "foo"
version = "0.0.1"

[[bin]]

name = "foo-exe"
path = "main.rs"
```

```
// main.rs

#![crate_type = "bin"]
pub fn main() {}
```

```
$ cargo --version
cargo 0.0.1-pre-nightly (0e42b4d 2015-03-25) (built 2015-03-25)
$ cargo build
failed to parse manifest at `/Users/dwrensha/Desktop/test-cargo/Cargo.toml`

Caused by:
  target names cannot contain hyphens: foo-exe
```

### PR DESCRIPTION
**Pull request by alexcrichton**
Examples, tests, binaries, etc, are all allowed to have hyphens (cargo does some
renaming internally).

Close #1450

### CODE REVIEW

**Comment by oferrigni** [comment]
You beat me. Was about to open a PR for https://github.com/oferrigni/cargo/commit/2c819c0108b780e30396dc003bfb69739faaab8f

File: src/cargo/util/toml.rs [hunk 1]
```diff
@@ -546,7 +534,7 @@ impl TomlManifest {
 
 fn validate_target_name(target: &TomlTarget) -> CargoResult<()> {
```
Comment (huonw) [inline src/cargo/util/toml.rs]: s/target/library/?

**Comment by huonw** [comment]
r=me with or without changing the function name (your choice).

(Travis failed, but it looks rather spurious.)

**Pull request merged**